code/modules/reagents/reagent_containers/blood_pack.dm

```diff
@@ -32,7 +32,7 @@
 		return TRUE
 
 	while(do_after(user, 1 SECONDS, timed_action_flags = IGNORE_USER_LOC_CHANGE))
-		if(!reagents.total_volume)
+		if(reagents.total_volume <= 0)
```

Comment:
this doesn't change anything

Reply:
Not true, if volume gets in the negatives somehow, it wont stop

Reply:
it will stop cause that is how the reagent procs work

Reply:
what would be a good idea would be to add an interaction key to the do afters

Reply:
i don't entirely know how interaction keys work, but i'll take a look

Reply:
sure enough, the do_after doesn't even target the bloodbag